### PR TITLE
refactor: move to single contenteditable

### LINKED ISSUE #1218: Unable to select text across blocks in Safari
https://user-images.githubusercontent.com/27926/219260686-ffc552ac-4b2a-43ed-8116-fdcfea90e885.mov

### PR DESCRIPTION
**Pull request by Flrande**
Close #1218

### CODE REVIEW

File: packages/lit/src/utils/range-binding.ts [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+    if (!to || PathFinder.equals(from.path, to.path)) return;
+
+    const range = this.rangeManager.value;
+    if (!range) return;
+
+    const blocks = this.rangeManager.getSelectedBlockElementsByRange(range, {
+      mode: 'flat',
+    });
+
+    const start = blocks.at(0);
```
Comment (Saul-Mirone) [inline packages/lit/src/utils/range-binding.ts:130]: Seems repeat with line 196

File: packages/lit/src/utils/range-manager.ts [hunk 2]
```diff
@@ -177,185 +131,126 @@ export class RangeManager {
...
+      },
+      to:
+        startBlock === endBlock
+          ? null
+          : {
+              path: endBlock.path,
+              index: endInlineRange.index,
+              length: endInlineRange.length,
+            },
+      isReverse: reverse,
```
Comment (Saul-Mirone) [inline packages/lit/src/utils/range-manager.ts:215]: Do we still need this property? I remember that this property was used to navigate with keyboard. Maybe we can try to remove it.

File: packages/lit/src/utils/range-binding.ts (hunk 1 above)
Comment (Flrande) [reply packages/lit/src/utils/range-binding.ts:135]: Do you mean them?

```ts
const start = blocks.at(0);
const end = blocks.at(-1);
if (!start || !end) return;

const startText = start.model.text;
const endText = end.model.text;
if (!startText || !endText) return;
```

File: packages/lit/src/utils/range-manager.ts (hunk 2 above)
Comment (Flrande) [reply packages/lit/src/utils/range-manager.ts:215]: Maybe we can use it to improve UX in remote-selection. 🤔

Comment (Saul-Mirone) [reply packages/lit/src/utils/range-manager.ts:215]: You mean the position of the name flag?

Comment (Flrande) [reply packages/lit/src/utils/range-manager.ts:215]: Yes

Comment (Flrande) [reply packages/lit/src/utils/range-manager.ts:215]: And I want to rename `isReverse` to `reverse` 🤪

File: packages/lit/src/utils/range-binding.ts (hunk 1 above)
Comment (Saul-Mirone) [reply packages/lit/src/utils/range-binding.ts:135]: yep

**Review by Saul-Mirone — APPROVED** [review]

**Comment by nawbc** [comment]
img_v3_027b_d05a63df-7bb1-415d-8f7e-1ea25aacaa9g

**Pull request merged**